nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = stdnse.strsplit("%;", split_str[#split_str])[1]
+    end
```

Comment:
I am not sure that the code always does what you intend. As an example, how about path `/users/joe.doe/welcome`?

Another possible issue could be path `/foo.`. Is this truly a zero-length extension or should this not be treated as an extension at all?

The code could be perhaps collapsed into the following one-liner:
```lua
parsed.extension = parsed.path:match("%.([^/;]+)%f[;\0][^/]*$")
```
...but definitely feel free to code it your way.